ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -266,7 +281,15 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		rayJobInstance.Status.JobDeploymentStatus = jobDeploymentStatus
```

Comment:
can we handle `jobDeploymentStatus` in the same place (i.e. `if isJobTerminal`)? We currently set `rayJobInstance.Status.JobDeploymentStatus` to `rayv1.JobDeploymentStatusComplete` or `rayv1.JobDeploymentStatusFailed`, and then do some other checks and update it to `rayv1.JobDeploymentStatusRetrying`.

```go
if isJobTerminal {
   // logic to determine the value of jobDeploymentStatus: rayv1.JobDeploymentStatusComplete, rayv1.JobDeploymentStatusFailed, or rayv1.JobDeploymentStatusRetrying
}
...
rayJobInstance.Status.JobDeploymentStatus = jobDeploymentStatus
```

Reply:
I don't think it can go under the `if isJobTerminal`, since isJobTerminal only checks the job status and not the job deployment status

Reply:
But I see your point about setting rayJobInstance.Status.JobDeploymentStatus multiple times, let me refactor to set it only once

Reply:
done